### DISCUSSION TITLE
Please share your CKEditor5 build method

### DISCUSSION THREAD

**Discussion by Nriver** [TriliumNext/Trilium#2824] [Q&A] [answered]
@zadam Would you please share your CKEditor5 plugins list or how you build ckeditor?

I am building a classic editor instead of the balloon editor used in the desktop client. It turns out that CKEditor will remove the html attrubutes/tags that it does not recognize. The editor ditches some of my note content when I call `editor.setData`, such as the todo list. I tried to include the `To-do list` plugin in my build and the to do list is woking now. But I am not sure about other plugins that I missed to include.

**Comment by zadam** [comment, marked as answer]
The build is here: https://github.com/zadam/trilium-ckeditor5/tree/stable/packages/ckeditor5-build-balloon-block

CKEditor 5 dropping unrecognized tag is its standard "feature". There is a (limited) way to get around it: https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html

**Reply by Nriver** [reply]
Thank you! I will have a good look at it.
